aws/cloudformation-templates/event-engine/segment-workshop.yaml

```diff
@@ -0,0 +1,149 @@
+    Default: 'dummy'
+
+  PreIndexElasticsearch:
+    Type: String
+    Description: >
+      Automatically index the Retail Demo Store products in Elasticsearch. Otherwise, select 'No' if you would 
+      prefer to complete this process yourself by stepping through the Search workshop included in this deployment 
+      as a Jupyter notebook in SageMaker.
+    AllowedValues:
+      - 'Yes'
+      - 'No'
+    Default: 'Yes'
+
+  PreCreatePersonalizeCampaign:
+    Type: String
+    Description: >
+      Automatically build solutions and launch Personalize campaigns. Otherwise, select 'No' if you would 
+      prefer to complete this process yourself by stepping through the Personalization workshop included in this deployment 
+      as a Jupyter notebook in SageMaker. Note that this process is done in the background after deployment finishes 
+      and can take 2+ hours to complete.
+    AllowedValues:
+      - 'Yes'
+      - 'No'
+    Default: 'No'
+
+  PreCreatePinpointWorkshop:
+    Type: String
+    Description: >
+      Automatically configure Pinpoint with messaging templates, segments, and campaigns. Otherwise, select 'No' if you would 
+      prefer to complete this process yourself by stepping through the Messaging workshop included in your deployment as a Jupyter notebook 
+      in SageMaker. Note that this option is dependent on Auto-Build Personalize Campaigns being enabled AND completed.
+    AllowedValues:
+      - 'Yes'
+      - 'No'
+    Default: 'No'
+
+  PinpointEmailFromAddress:
+    Type: String
+    Description: >
+      Email address to use as Reply-To when sending emails from Pinpoint. This parameter is required when 'Auto-Configure Pinpoint'
+      is set to 'Yes'.
+    Default: 'dummy@dummy.com'
+  
+  PinpointEmailFromName:
+    Type: String
+    Description: >
+      Name to use with Reply-To email address when sending emails from Pinpoint. This parameter only applies when 'Auto-Configure Pinpoint'
+      is set to 'Yes'.
+    Default: "AWS Retail Demo Store"
+  
+  AmplitudeApiKey:
+    Type: String
+    Description: Amplitude API key for integrated product analytics and A/B testing results (optional).
+    NoEcho: true
+    Default: NONE
+
+  OptimizelySdkKey:
+    Type: String
+    Description: Optimizely SDK key for experimentation (optional).
+    NoEcho: true
+    Default: NONE
+
+  SegmentWriteKey:
+    Type: String
+    Description: Segment write key for real-time data collection (optional).
+    NoEcho: true
+    Default: NONE
+
+Resources:
+  # This references the base template.yaml file that comes with the Retail Demo Store.  Set the default values
+  # for the properties above to whatever you need to set up your environment in the Event Engine.  Happy Eventing. 
+  EventEngineRetailDemoStore:
+    Type: AWS::CloudFormation::Stack
+    Properties:
+      TemplateURL: !Sub https://s3.amazonaws.com/${ResourceBucket}/${ResourceBucketRelativePath}cloudformation-templates/template.yaml
+      Parameters:
```

Comment:
All the parameters will need to be listed here so they're passed to template.yaml. So the precreate and GH params.